### PR TITLE
#1612 - GTM configuration

### PR DESCRIPTION
**Pull request by maciekz1996**
**Changes**
- Installed `react-gtm-module` package
- Configured Google Tag Manager and added custom events
- Fixed bug while signing in/up with email and password

**Custom events that are now passed to Google Analytics**
- `sign_up` - whenever a new user registers to the Curator portal
    * `method` - method of registration (`google` or `email`)
    * `newsletter_accepted` - whether a user accepted the newsletter
- `full_dataset_download` - fired when a full dataset download occurs
- `filtered_dataset_download` - fired when a partial dataset download occurs
    * `download_format` - chosen data format
    * `query` - applied query
- `filters_applied` - fired when some filters are being applied
    * `query` - applied query

### CODE REVIEW

File: verification/curator-service/ui/src/index.tsx [hunk 1]
```diff
@@ -10,6 +10,13 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import store from './redux/store';
 import { Provider } from 'react-redux';
+import TagManager from 'react-gtm-module';
+
+const tagManagerArgs = {
+    gtmId: 'GTM-WHCJVVH',
```
Comment (abhidg) [inline verification/curator-service/ui/src/index.tsx:16]: Checking that this is fine to be public @maciekz1996 ?

Comment (iamleeg) [reply verification/curator-service/ui/src/index.tsx:16]: Yes, according to [the setup guide](https://support.google.com/tagmanager/answer/6103696?hl=en#zippy=%2Cweb-pages) you put a JS script with your tag ID into the published HTML, and I can see other sites with their GTM ID published too, this is fine.

**Review by iamleeg — APPROVED** [review]

**Comment by abhidg** [comment]
@maciekz1996 there’s a conflict - will merge on resolve.

**Review by iamleeg — APPROVED** [review]
Thanks @maciekz1996!

**Pull request merged**